packages/core-data/src/footnotes/index.js

```diff
@@ -56,30 +61,20 @@ export function updateFootnotesFromMeta( blocks, meta ) {
 				continue;
 			}
 
-			// When we store rich text values, this would no longer
-			// require a regex.
-			const regex =
-				/(<sup[^>]+data-fn="([^"]+)"[^>]*><a[^>]*>)[\d*]*<\/a><\/sup>/g;
-
-			attributes[ key ] = value.replace(
-				regex,
-				( match, opening, fnId ) => {
-					const index = newOrder.indexOf( fnId );
-					return `${ opening }${ index + 1 }</a></sup>`;
-				}
-			);
-
-			const compatRegex = /<a[^>]+data-fn="([^"]+)"[^>]*>\*<\/a>/g;
-
-			attributes[ key ] = attributes[ key ].replace(
-				compatRegex,
-				( match, fnId ) => {
-					const index = newOrder.indexOf( fnId );
-					return `<sup data-fn="${ fnId }" class="fn"><a href="#${ fnId }" id="${ fnId }-link">${
-						index + 1
-					}</a></sup>`;
-				}
-			);
+			const richTextValue = create( { html: value } );
+
+			richTextValue.replacements.forEach( ( replacement ) => {
+				if ( replacement.type !== 'core/footnote' ) return;
+
+				const id = replacement.attributes[ 'data-fn' ];
+				const index = newOrder.indexOf( id );
+
+				replacement.innerHTML = `<a href="#${ id }" id="${ id }-link">${
+					index + 1
+				}</a>`;
```

Comment:
I suppose that there's nothing we need to preserve from `replacement` here? that we're clear to completely replace its contents?

Reply:
No, I added a comment pointing to 

https://github.com/WordPress/gutenberg/blob/6db51a6ee92db6c3324a875135656861bbd9e55d/packages/block-library/src/footnotes/format.js#L108

Reply:
I'd rather not recreate the HTML, but then we still need to parse HTML... I guess we could use rich text `create` here too though. 🤔

Reply:
I didn't mean to communicate more than a question here, that in completely replacing the inner HTML we didn't need to remember to bring something that we previously had there.

the text-HTML-construction is fine given that we parsed the integers and know we'll have safe strings.

the changes look fine of course, so there's no reason they need to go back to the string construction.